src/core/models/outputs/output_extended.c

```diff
@@ -11,36 +11,59 @@
 #include "utlist.h"
 
 // minimum dust allowance
-#define MIN_DUST_ALLOWANCE 1000000
+// #define MIN_DUST_ALLOWANCE 1000000
 
 // maximum number of feature blocks
 #define MAX_FEATURE_BLOCKS_COUNT 8
+// maximum number of unlock condition blocks
+#define MAX_EXTENDED_CONDITION_BLOCKS_COUNT 4
+// maximum number of feature blocks
+#define MAX_EXTENDED_FEATURE_BLOCKS_COUNT 3
 
 output_extended_t* output_extended_new(address_t* addr, uint64_t amount, native_tokens_t* tokens,
-                                       feat_blk_list_t* feat_blocks) {
-  if (addr == NULL) {
+                                       cond_blk_list_t* cond_blocks, feat_blk_list_t* feat_blocks) {
+  if (addr == NULL || cond_blocks == NULL) {
     printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
     return NULL;
   }
 
-  if (amount < MIN_DUST_ALLOWANCE) {
-    printf("[%s:%d] dust allowance amount must be at least 1Mi\n", __func__, __LINE__);
+  // FIXME: amount must fulfill the Byte Cost Dust Protection
+  // if (amount < MIN_DUST_ALLOWANCE) {
+  //   printf("[%s:%d] dust allowance amount must be at least 1Mi\n", __func__, __LINE__);
+  //   return NULL;
+  // }
+
+  if (cond_blk_list_len(cond_blocks) > MAX_EXTENDED_CONDITION_BLOCKS_COUNT) {
+    printf("[%s:%d] there should be at most %d condition blocks\n", __func__, __LINE__,
+           MAX_EXTENDED_CONDITION_BLOCKS_COUNT);
     return NULL;
+  } else {
+    // must no contain UNLOCK_COND_STATE or UNLOCK_COND_GOVERNOR
+    if (cond_blk_list_get_type(cond_blocks, UNLOCK_COND_STATE) ||
+        cond_blk_list_get_type(cond_blocks, UNLOCK_COND_GOVERNOR)) {
+      printf("[%s:%d] State Controller/Governor conditions are not allowed\n", __func__, __LINE__);
+      return NULL;
+    }
   }
 
   if (feat_blk_list_len(feat_blocks) > MAX_FEATURE_BLOCKS_COUNT) {
```

Comment:
Should `MAX_FEATURE_BLOCKS_COUNT` be replaced with `MAX_EXTENDED_FEATURE_BLOCKS_COUNT`?